### PR TITLE
[dsme] Add enablers for blocking shutdown via D-Bus. JB#62532

### PR DESCRIPTION
**Pull request by spiiroin**
If device is powered off while startup wizard is running, glitching
1st boot actions can cause persistent problems on subsequent bootups.

There is little that can be done about forced hw power offs. But by
having dbus methods available for blocking soft shutdown / reboot
attempts we can at least decrease the likelihood of such problems.

Only privileged processes can block shutdown / reboot.

Blocking does not prevent thermal / battery empty shutdown.

### CODE REVIEW

File: rpm/dsme.spec [hunk 1]
```diff
@@ -10,13 +10,15 @@ Source2:    dsme-rpmlintrc
 Requires:   systemd
 Requires:   ngfd
 Requires:   libdsme >= 0.66.0
+#                      ^^^^^^ TBD
```
Comment (spiiroin) [inline rpm/dsme.spec]: Reminder: version deps need to be updated before merging

File: util/waitfordsme.c [hunk 2]
```diff
@@ -48,7 +48,7 @@ static bool send_to_dsme(dsmesock_connection_t* conn, const void* msg)
 }
 
 int main(int argc, char* argv[]) {
-	
+
```
Comment (pvuorela) [inline util/waitfordsme.c]: Could as well remove here in the beginning of the function.

**Review by pvuorela — COMMENTED** [review]
Quite a bit changes here, but can't at least point anything clearly wrong. Had elsewhere question about naming so not immediately approving.

File: util/waitfordsme.c (hunk 2 above)
Comment (spiiroin) [reply util/waitfordsme.c]: Now all of that too should be gone.

**Review by pvuorela — APPROVED** [review]

File: rpm/dsme.spec (hunk 1 above)
Comment (spiiroin) [reply rpm/dsme.spec]: done

**Pull request merged**